src/md/compatibility.md

```diff
@@ -0,0 +1,112 @@
+
+Policy
+
+The REST API is separately versioned. This is to allow for co-existence of old and new API should there be a need for backward incompatible changes in the future.
+
+###Command Line Interface (CLI)
+
+The CLI may be used either directly via the system shell or via shell scripts. Changing the path, removing or renaming command line options, the order of arguments, or the command return code and output break compatibility and may adversely affect users.
+
+Policy
+
+CLI commands are to be deprecated (warning when used) in a prior minor release before they are removed or incompatibly modified in a subsequent major release.
+
+###Configuration Files
+
+Configuration files are used for engine or application settings. Changes to keys and default values affect users. 
+
+Policy
+
+Name, location, format, keys of configuration files should be deprecated in a prior minor release and can only be changed in major release. Best effort should be made to support the deprecated behavior for one more major release (not guaranteed). 
+
+###Internal Wire compatibility
+
+Apex containers internally use RPC communication and netlet for the data flow. The protocols are private and user components are not exposed to it. Apex is a YARN application and automatically deployed. There is currently no situation where containers of different Apex engine versions need to be interoperable. Should such scenario become relevant in the future, wire compatibility needs to be specified.
+
+Policy
+
+N/A
+
+###Internal File formats
+
+Apex engine stores data in the file system for recovery. When using serialization (Kryo, Java etc.), changes to internal classes may affect the ability to relaunch an application with upgraded engine code from previous state. This is currently not supported. In the future, the serialization mechanism should guarantee backward compatibility.
+
+Policy
+
+Currently no compatibility guarantee. User to cold-restart application on engine upgrade.
+
+###Java Classpath
+
+Apex applications should not bundle Hadoop dependencies or Apex engine dependencies but use the dependencies provided in the target environment to avoid conflicts. The Apex application archetype can be used to generate a compliant project.  
+
+Policy
+
+Apex engine dependencies can change as per semantic versioning.
+
+###Maven Build Artifacts
+
+Downstream projects reference the Apex engine dependencies and Malhar operator libraries for application development etc. Changes to the packaging (which classes are in which jar), the groupId, artifactId and which artifacts are deployed to Maven central impact upgrades.
+
+Policy
+
+The artifacts that contain the classes that form the public API as specified above cannot change on patch releases and should stay compatible within a major release.
```

Comment:
If we don't even allow patch version change in the dependency artifacts, maybe we should explicitly say that?